### ISSUE TITLE
mypy fails to recognize functions as generators when `sys.platform` condions are used

### ISSUE THREAD

**Issue by marcinsulikowski** [python/mypy#5678]
`mypy` seems to be too thorough when interpreting `if sys.platform == ...` conditions. Consider the following file:

```py
import sys
from typing import Generator

def f() -> Generator[int, None, None]:
    if sys.platform == "darwin":
        yield 1
```
I'm trying to run `mypy` for this file on Linux using `mypy gen.py`

**Expected behavior:**
`mypy` test passes for this file on Linux

**Actual behavior:**
I get this on Linux:
```
$ mypy gen.py 
gen.py:4: error: Missing return statement
```
At the same time, `mypy` on macOS does not complain. I think that this is related to #698

**Platform information:**
```
$ uname -sr; mypy --version; python --version
Linux 4.15.0-34-generic
mypy 0.630
Python 2.7.15rc1
```

**Comment by gvanrossum** [comment]
Agreed. It looks like the analysis that determines whether a function body contains any `yield`s runs after the version/platform-based dead code elimination; it ought to run before that. Are you interested in trying to fix this?
